### ISSUE TITLE
Cannot read properties of undefined (reading 'css') (resize.js:64)

### ISSUE THREAD

**Issue by reklamasieciowa** [doxygen/doxygen#11000]
**Describe the bug**
Resize.js attempts to assign styles to the navtree element.  If `GENERATE_TREEVIEW = NO` is in the configuration, the element does not exist.
```
navtree = $("#nav-tree");
[...]
navtree.css({height:navtreeHeight + "px"});
```
**Screenshots**
![image](https://github.com/user-attachments/assets/5ca096ed-29bc-4a2f-b79f-3e3fbced68b3)

**To Reproduce**
Set 
```
GENERATE_TREEVIEW = NO
DISABLE_INDEX = NO
```
Check the errors in the console.

**Expected behavior**
Checking if an element exists in the DOM.

**Version**
1.11.0

**Comment by albert-github** [comment]
I tried to reproduce the problem but was unable to reproduce it
- which browser + version are you using?
- what exact steps did you take to get to the error
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by reklamasieciowa** [comment]
1. Google Chrome 126.0.6478.127 (64-bit) and Firefox 128.0 (64-bit).
2. The error appeared after Doxygen 1.10.0 -> 1.11.0 update.
3. In the code, you can see the new ID of the content container and the added height. We don't see this in version 1.10.0.
<div id="doc-content" style="height: 673px;">.
4. Unfortunately, I can't provide the source files, as this is internal documentation. I can only provide the configuration file.

5. If there is a possibility, it would be very nice to have a configuration option, such as: DISABLE_RESIZE = YES. ;)
![image](https://github.com/user-attachments/assets/32885cba-b8a3-48be-8942-411d0a4f15ef)

**Comment by albert-github** [comment]
> 4. Unfortunately, I can't provide the source files, as this is internal documentation. I can only provide the configuration file.

Mockup / dummy code would be nice .s necessary ...

My example: [example.tar.gz](https://github.com/user-attachments/files/16232770/example.tar.gz) and I tested with FireFox 128.0 (64-bit) as well.

In the doxygen 1.11.0 version I see a definition for the `#doc-content` in the files `doxygen.css` and `navtree.css`, in the doxygen 1.10.0 version only in the `doxxygen.css` (there is no navtree.css` in the 1.10.0 version.
In the doxygen 1.10.0 version I don't see the `resize.js` 

Did a bisect and the adding of the files looks like to occur in (109afe13c3f9accaf2dbae318341613cba5e897f):
```
109afe13c3f9accaf2dbae318341613cba5e897f is the first bad commit
commit 109afe13c3f9accaf2dbae318341613cba5e897f

Date:   Fri Jan 5 10:22:35 2024 +0100

    Make the menu bar stick to the top of the screen when scrolling

    In the past an attempt was made (see fcb8cfbcd67d95680ccde950c03fd7e67acdb5b3) to get the top part of the screen fixed in case of `GENERATE_TREEVIEW=NO`.
    This method was not working and after a small time has been reverted.
    In the current implementation:
    - adds a tag to the HTML file in case of `GENERATE_TREEVIEW=NO` (`div id="doc-content">`) that was also used in case of `GENERATE_TREEVIEW=YES` so `resize.js` can handle the scrollbar
    - add end tag to the `footer.html` in case of `GENERATE_TREEVIEW=NO`
    - always use `resize.js` and `navtree.css`
    - adjust `resize.js` so it can be used for the `GENERATE_TREEVIEW=NO` case (also some javascript arrow functions, `=>` changes had to be reverted for CHM generation).

 src/ftvhelp.cpp            | 22 -------------
 src/htmlgen.cpp            | 56 ++++++++++++++++++++++++++------
 templates/html/footer.html |  1 +
 templates/html/menu.js     |  4 +--
 templates/html/resize.js   | 81 ++++++++++++++++++++++++++++------------------
 5 files changed, 99 insertions(+), 65 deletions(-)

```

Looks that besides the dependency `GENERATE_TREEVIEW` there might also be a dependency necessary for `DISABLE_INDEX`, but as I cannot reproduce the problem it is hard to tell (@reklamasieciowa so a small example would be quite bnecessary).

@doxygen any idea?

**Comment by albert-github** [comment]
@reklamasieciowa 
I see in the `resize.js` the line 51:
```
    if (treeview)
```
so the part of the code should already be protected.
- What is the value of `treeview`? 
  Just place a `alert(treeview)` just in front of the `if` statement.
  (In my case this I get `false`).

**Comment by reklamasieciowa** [comment]
![image](https://github.com/user-attachments/assets/40799cb2-9b23-4aa8-a35a-fe74ff494fdb)

**Comment by albert-github** [comment]
That is strange.
- Are you using your own `header.html` ?
  - Please provide the entire `Doxyfile` you used.
- Did you try the example I provided? 
   (just extract the file and run doxygen on it.

**Comment by reklamasieciowa** [comment]
1. Yes, we are using custom header.
2. Yes, example build has the same error.
![image](https://github.com/user-attachments/assets/5997056c-12f4-41c4-8f9b-9551de8d485b)

**Comment by albert-github** [comment]
As the small example exhibits at your place the same problem, can you provide the resulting html tree of the small example?
When I run I get: [example_html.tar.gz](https://github.com/user-attachments/files/16233960/example_html.tar.gz)

**Comment by reklamasieciowa** [comment]
Of course, html output:
[html.zip](https://github.com/user-attachments/files/16234242/html.zip)

**Comment by reklamasieciowa** [comment]
Ok, I downloaded your output and in it I also have an error in Chrome, but not in Firefox. Maybe it's one of the plugins. :)

**Comment by albert-github** [comment]
Your HTML is identical to the HTML I send so this can be excluded.
I checked  with the chrome version you mentioned as well and I don't get the error.
You already mentioned that you disabled the Chrome plugins, how about the extensions?
(normally I use FireFox, so a bit less experienced with Chrome. I also use the English, US, version so maybe the Polish version ...., but this is highly unlikely).

All in all, to be honest, very very strange and beats me.

**Comment by albert-github** [comment]
Even though "beats me" some thoughts:
- which page, probably all as it happens with your own normal work and also with my small example.
- cache effect, probably not as you also see it in "anonymous mode". Never hurts to try a "forced" reload of the page (From a google search for "reload page with clear cache": Open the Chrome Dev Tools by pressing F12. Once the Chrome Dev Tools are open, right-click on the browser Reload button and choose Empty Cache and Hard Reload. Alternatively, hold down the Ctrl key and click the Reload button. Alternatively, hold down the Ctrl key and press F5)
- last resort go into debugging mode and do a step by step load and try to pinpoint where the wrong "initMenu" is called (at least I suspect that it is called in the wrong way somewhere, though the code (the small  example) looks all consistent and OK to me.

**Comment by reklamasieciowa** [comment]
Well, it took a while, but I managed to debug the problem. 
The resizeHeight function is called three times. Twice it receives the parameter false. The problem is the third call on onload:
https://github.com/doxygen/doxygen/blob/master/templates/html/resize.js#L143
Calling the function without parentheses, for some reason, passes the event itself to it. This, of course, causes the parameter to be an object, which is true.
![image](https://github.com/user-attachments/assets/e1f94b0c-2be6-4bfb-912b-5c5a13c63c53)

![image](https://github.com/user-attachments/assets/63f7097a-2f15-4a93-9f76-175543200a8b)

A simple fix: calling resizeHeight with parentheses -> resizeHeight() or with parameter resizeHeight(treeview) eliminates this problem.
It works in both Chrome and FF.
![image](https://github.com/user-attachments/assets/48d4f194-0921-4b07-a89e-faf27e7eb987)

**Comment by doxygen** [comment]
@albert-github Looks like this was introduced by https://github.com/doxygen/doxygen/pull/10537

@reklamasieciowa  Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by reklamasieciowa** [comment]
Yes, everything looks fine.

By the way, we need to override the height set in inline styles for #doc-content anyway, because it makes obnoxious scrolled frames. ;)

**Comment by albert-github** [comment]
@reklamasieciowa thanks for the debugging and testing.

Makes sense. I remember (very vaguely) that I had some problems finding the right definition for the `on` method, didn't see an example with an argument (in the `function` part). In my previous comments I concentrated a bit to much on the `initMenu` as it was clear from the outside that the signature had been changed between 1.10.0 and 1.11.0 but that was all OK. In the images (e.g. in https://github.com/doxygen/doxygen/issues/11000#issue-2408228861) the stack trace indicates indeed `resizeHeigh` but I missed it.

A bit strange remains for me why I only see twice the `false` and @reklamasieciowa sees twice the `false` and at the third instance ((https://github.com/doxygen/doxygen/issues/11000#issuecomment-2229107198) with the "uncaught".

**Comment by reklamasieciowa** [comment]
Są różnice miedzy wywołaniem onload pomiędzy Chrome i FF, ale nie mam zacięcia, żeby badać szczegóły. ;) 
I also had the error in FF, but maybe the wrong function call was not running in your case, and in my case it passed the event as a parameter.
https://www.google.com/search?q=%24(window).on(%27load%27+firefox+problem&oq=%24(window).on(%27load%27+firefox+problem&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIJCAEQIRgKGKAB0gEINzM3NWowajeoAgCwAgA&sourceid=chrome&ie=UTF-8

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
